### ISSUE TITLE
Server Returns Detailed Error Messages

### ISSUE THREAD

**Issue by andreeaursu1981** [CityOfBoston/digital#614]
EY identified that the GraphQL endpoint returned detailed error messages to clients regarding the status of application exceptions when an exception occurred. During the testing, EY tampered with the parameters on the endpoint https://access.boston.gov/graphql that resulted in an exception error in the application.
To reproduce, follow the steps below:Configure your browser to use a proxy tool, such as Burp.Navigate to https://access.boston.gov and authenticate with valid account credentials.Select MFA device and retrieve email “Access Boston Security Code” or SMS code sent to selected device.Enter received MFA passcode on PingID authentication page.Observe user account is successfully logged in to Access Boston application home page.Next, click “Change My Password” URL found under “Account Tools” section on Access Boston application home
page.Enter in current user’s account password and populate new password and confirm password form fields and click
“Change Password”After submitting the above request in step 7, intercept the HTTP POST request sent to the
https://access.boston.gov/graphql endpoint in Burp’s “Proxy” tab.Change the POST request method to GET and modify the “/graphql” parameter in the URL with
/graphql?query={_schema} and click “Forward”Intercept the HTTP response returned by the server in Burp’s “Proxy” tab.In the Proxy tab window, observe intercepted HTTP response contained error message that resulted in information
leakage of the underlying Apollo Server technology.

**Comment by reillyzlab** [comment]
Sent via email by @phillipbentonkelly 10/20/20 at 11:45 a.m.:

I created a patch for this issue and got it working on my computer. I deployed it to the DEV environment yesterday morning but at the moment I am unable to verify if it working because I don't have the ability to install software on my remote desktop. I pinged doit help desk to give me access but I've been waiting two days for them to get to it. Andreea, if someone on your team can follow the instructions on the doc and install Burp or Postman to test in the meantime it would help me. The patch I did changes the message you get when there's an error to a generic message with no internal structure data. In addition, would we want to keep our meeting later today or could we reschedule until I/someone is able to test on the DEV environment?
